test/src/test/java/hudson/search/SearchTest.java

```diff
@@ -105,6 +105,40 @@ public void testSearchByProjectName() throws Exception {
         assertTrue(contents.contains(String.format("<title>%s [Jenkins]</title>", projectName)));
     }
 
+    @Issue("JENKINS-24433")
+    @Test
+    public void testSearchByProjectNameBehindAFolder() throws Exception {
+        final String projectName = "testSearchByProjectName";
+        j.createFreeStyleProject(projectName);
+        j.createFolder("my-folder-1").createProject(FreeStyleProject.class, "my-job-1");
+
+        Page result = j.createWebClient().goTo("job/my-folder-1/search?q="+ projectName);
+
+        assertNotNull(result);
+        j.assertGoodStatus(result);
+
+        // make sure we've fetched the testSearchByDisplayName project page
+        String contents = result.getWebResponse().getContentAsString();
+        assertTrue(contents.contains(String.format("<title>%s [Jenkins]</title>", projectName)));
+    }
+
+    @Issue("JENKINS-24433")
+    @Test
+    public void testSearchByProjectNameInAFolder() throws Exception {
+        final String projectName = "testSearchByProjectName";
+        j.createFreeStyleProject(projectName);
+        j.createFolder("my-folder-1").createProject(FreeStyleProject.class, "my-job-1");
+
+        Page result = j.createWebClient().goTo("job/my-folder-1/search?q="+ "my-folder-1/my-job-1");
+
+        assertNotNull(result);
+        j.assertGoodStatus(result);
+
+        // make sure we've fetched the testSearchByDisplayName project page
```

Comment:
In fact you are getting `my-job-1`, not `testSearchByProjectName`. Unclear why `testSearchByProjectName` even needs to be created in this test case.